rules.bzl

```diff
@@ -52,11 +52,16 @@ def _linker_override(ctx, override_linkopts):
         linkopts.extend(ctx.attr.ld64_linkopts)
 
     linkopts_depset = depset(direct = linkopts, order = "topological")
+
+    objc_provider_kwargs = {}
+    if hasattr(apple_common.new_objc_provider(), "linkopt"):
```

Comment:
While not as bad as the rules_swift suggestion, I think it would be worth making this globally.